olive/model/__init__.py

```diff
@@ -56,68 +57,61 @@ def __init__(
+        resources["model_path"] = model_path
+        for resource_name, resource_path in resources.items():
+            self.resource_paths[resource_name] = create_resource_path(resource_path)
+            # initialize local resource paths to None
+            self.local_resource_paths[resource_name] = None
 
     @property
     def model_path(self) -> str:
         """Return local model path."""
-        # return None if model_path is None
-        if self.model_resource_path is None:
-            return None
-
-        # return local path if model_path is local or string name
-        if self.model_resource_path.is_local_resource() or self.model_resource_path.is_string_name():
-            return self.model_resource_path.get_path()
-
-        # check if model_path is downloaded
-        if self.local_model_path is None:
-            raise ValueError(
-                "Model is not local and not downloaded yet. Please call download_model() or set_local_model_path()"
-                " first."
-            )
-        return self.local_model_path.get_path()
+        return self.get_local_resource("model_path")
 
-    def download_model(self, download_path: Union[Path, str], overwrite: bool = False) -> str:
+    def get_local_resource(self, resource_name: str) -> str:
         """
-        Download model to local path.
+        Get local path of a resource.
 
-        :param download_path: local directory to download model to.
-        :param overwrite: whether to overwrite existing model. If not overwrite, will raise exception if model exists.
-        :return: local model path.
+        :param resource_name: name of the resource.
+        :return: local path.
         """
-        # return None if model_path is None
-        if self.model_resource_path is None:
-            logger.debug("Model path is None, skip downloading.")
-            return None
-
-        # return local path if model_path is local or string name
-        if self.model_resource_path.is_local_resource() or self.model_resource_path.is_string_name():
-            logger.debug("Model path is local or string name, skip downloading.")
-            return self.model_resource_path.get_path()
-
-        # download model
-        logger.debug(f"Downloading model to {download_path}")
-        self.local_model_path = create_resource_path(
-            self.model_resource_path.save_to_dir(download_path, overwrite=overwrite)
-        )
-        return self.local_model_path.get_path()
+        assert resource_name in self.resource_paths, f"{resource_name} is not a valid resource name."
+        resource_path = self.resource_paths[resource_name]
+        local_resource_path = self.local_resource_paths[resource_name]
+        local_path = resolve_local_resource(resource_path, local_resource_path)
+        if not local_path and resource_path:
+            raise ValueError(
+                f"{resource_name} is not local and not downloaded yet. Please call download_model() or"
```

Comment:
>download_model [](http://example.com/codeflow?start=83&length=14)

download_model is removed?

Reply:
Yes, like we discussed offline, the `download_model` method was never used. We also want to handle the model upload,download outside this model, so I removed this method.